feature/google/src/latestDependencies/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -155,26 +154,47 @@ class BillingWrapper(
         log(LogIntent.DEBUG, OfferingStrings.FETCHING_PRODUCTS.format(skus.joinToString()))
         executeRequestOnUIThread { connectionError ->
             if (connectionError == null) {
-                val params = SkuDetailsParams.newBuilder()
-                    .setType(productType.toGoogleProductType() ?: SkuType.INAPP)
-                    .setSkusList(nonEmptySkus).build()
+                val productList = skus.map { sku ->
+                    QueryProductDetailsParams.Product.newBuilder()
+                        .setProductId(sku)
+                        .setProductType(productType.toGoogleProductType() ?: BillingClient.ProductType.SUBS)
+                        .build()
+                }
+                val params = QueryProductDetailsParams.newBuilder()
+                    .setProductList(productList).build()
 
                 withConnectedClient {
-                    querySkuDetailsAsyncEnsuringOneResponse(params) { billingResult, skuDetailsList ->
+                    queryProductDetailsAsyncEnsuringOneResponse(params) { billingResult, productDetailsList ->
                         if (billingResult.responseCode == BillingClient.BillingResponseCode.OK) {
                             log(
                                 LogIntent.DEBUG, OfferingStrings.FETCHING_PRODUCTS_FINISHED
                                     .format(skus.joinToString())
                             )
                             log(
                                 LogIntent.PURCHASE, OfferingStrings.RETRIEVED_PRODUCTS
-                                    .format(skuDetailsList?.joinToString { it.toString() })
+                                    .format(productDetailsList.joinToString { it.toString() })
                             )
-                            skuDetailsList?.takeUnless { it.isEmpty() }?.forEach {
-                                log(LogIntent.PURCHASE, OfferingStrings.LIST_PRODUCTS.format(it.sku, it))
+                            productDetailsList.takeUnless { it.isEmpty() }?.forEach {
+                                log(LogIntent.PURCHASE, OfferingStrings.LIST_PRODUCTS.format(it.productId, it))
+                            }
+
+                            val storeProducts = mutableListOf<StoreProduct>()
+                            productDetailsList.forEach { product ->
+                                val basePlans = product.subscriptionOfferDetails?.filter {
+                                    it.pricingPhases.pricingPhaseList.size == 1
+                                } ?: emptyList()
+
+                                val groupedOffers = product.subscriptionOfferDetails?.groupBy {
+                                    it.pricingPhases.pricingPhaseList.last().toRevenueCatPricingPhase()
```

Comment:
Minor: I'd probably use the billingPeriod of the last pricingPhase instead of the full pricingPhase object

Reply:
Oh I think that makes sense yes. There won't be another base plan with the same duration so just comparing durations should work

Reply:
Hmmm thinking about it, is it possible that 2 base plans have same billing period but different billingCycleCount? I can test it

Reply:
<img width="699" alt="Screen Shot 2022-10-28 at 5 35 45 PM" src="https://user-images.githubusercontent.com/664544/198676938-6d0226ad-bac3-49ab-9ac3-436c01f79051.png">

ok! not possible to create two base plans with same billing period